### PR TITLE
Delete items from my shopping list, error made or don't need item #12

### PR DESCRIPTION
**Pull request by AshlingH**
## Description

Delete an Item: Create a button function to delete items

Prompt the User: send a modal to User and ask " Do you really want to delete this item?"

Deletion: Access database and remove delete item from database

## Related Issue
#13 

## Acceptance Criteria

-User is able to delete an item from the shopping list
-Prompt the user to confirm that they really want to delete the item to prevent accidental deletions
-Deletion should cause the associated record(s) in the database to be deleted
 
## Type of Changes

 | Type
-- | --
  | 🐛 Bug fix
✓ | ✨ New feature
✓ | 🔨 Refactoring
  | 💯 Add tests
  | 🔗 Update dependencies
  | 📜 Docs

## Thoughts

Some very interesting discussions about what to create and where to create it. We decided on creating a component Modal.js and keeping the deleteItem logic in ShoppingList.

**Roadblocks included:**

 -In the return using the appropriate methods to call our modal and also call both a "Yes" and "No" button within the modal.

-Acepting that the use of Props was a good way to tackle this challenge and also read up on props again!!

### CODE REVIEW

File: src/pages/ShoppingList.js [hunk 1]
```diff
@@ -91,61 +103,82 @@ const ShoppingList = ({ token }) => {
...
+            .doc(item.id)
+            .delete()
+            .then(() => (getShoppingList(), setModal(false)));
+    };
+
     const filteredList = shoppingListItems.filter(item => {
         return item.itemName.toLowerCase().includes(filterString.toLowerCase());
     });
-
+    console.log('Modal status: ', modal);
```
Comment (joeylaguna) [inline src/pages/ShoppingList.js]: I think we can remove this if we're done testing 😄

Comment (jonbascos) [reply src/pages/ShoppingList.js]: Done.  Just pushed the update!

File: src/components/AddItemForm.js [hunk 2]
```diff
@@ -59,6 +59,7 @@ const Form = ({ token }) => {
             .replace(/\s{2,}/g, ' ');
         if (!shoppingListCollection.includes(normalizeItemName)) {
             let data = {
+                numOfPurchases: 0,
                 itemName,
```
Comment (laurenmbeatty) [inline src/components/AddItemForm.js:63]: 👍 ... good add.

File: src/components/Modal.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    return (
+        <div className="Modal">
+            <p> Are you sure you want to delete {props.item.itemName}?</p>
+            <button onClick={() => props.delete(props.item)}>YES</button>
+            <button onClick={() => props.cancel()}>No</button>
+        </div>
+    );
+};
+
+export default Modal;
```
Comment (laurenmbeatty) [inline src/components/Modal.js:14]: Glad you made this its own component in a separate file. Nice and clean.

**Comment by laurenmbeatty** [comment]
This works great...even if I filter it down to just one item, and delete it...nothing fails!  Well done. 💪

**Review by dkimlim — APPROVED** [review]
Worked well on my end when I pulled the branch! My only small comment is maybe moving the modal closer to the item being deleted? I almost missed it at the top of the page, but again, this is just a minor comment! Good job on this new component :D

**Review by laurenmbeatty — APPROVED** [review]
Ship it! 🚢

**Review by jendevelops — APPROVED** [review]
Awesome y'all!

**Review by joeylaguna — APPROVED** [review]

**Pull request merged**